### PR TITLE
Feat: Recaptcha validation

### PR DESCRIPTION
**Pull request by Meldiron**
Hey there 👋 I have created an implementation for Google reCAPTCHA V2 so we can start the code review process.

When we agree on the structure, let's keep the PR open and I will implement more providers before the final review and merge.

I did manual QA with solution in this PR and was able to get it working on my demo application:

CleanShot 2022-06-12 at 10 47 29@2x

CleanShot 2022-06-12 at 10 47 18@2x

Related to issue https://github.com/Basharath/FormEasy/issues/1

### CODE REVIEW

**Comment by Meldiron** [comment]
One concern you might have is why I named the parameter `g-recaptcha-response` (in readme, step 6. also Code.js line 105).

This name comes from Google Docs, and while we could easily rename it to something simpler, I find it important to stay as close to the native solution as possible. That will help people experienced with Google reCAPTCHA to easily use it in FormEasy without having to scan docs.

On the other hand, this will create inconsistency between different captcha providers. This could be a valid reason to try to use specific words for all of them, like `captcha`. But with that, if some provider requires more than just a string response, we would make it a mess and people would have to learn how to pass it into FormEasy.

---

I would love to hear your feedback on this.

File: README.md [hunk 1]
```diff
@@ -98,6 +98,58 @@ Note: The keys of the `data` object should match with the fields that are set us
...
 
+## Captcha validation
+
+FormEasy supports multiple captcha providers to allow you to prevent unverified submissions by robots. Each provider is unique and requires a unique configuration. Please refer to the documentation below to enable a specific captcha provider.
+
+### Google reCAPTCHA V2
+
+1. Register a site and get your secret key, and site key: [https://www.google.com/recaptcha/admin/create](https://www.google.com/recaptcha/admin/create)
+
+2. In your apps script file, inside function `doPost, add the following configuration:
```
Comment (Basharath) [inline README.md]: You missed adding ` at the end of doPost.

File: README.md [hunk 2]
```diff
@@ -98,6 +98,58 @@ Note: The keys of the `data` object should match with the fields that are set us
...
+</head>
+```
+
+4. Add reCAPTCHA input into your form:
+
+```html
+<div class="g-recaptcha" data-sitekey="YOUR_SITE_KEY"></div>
+```
+
+> Make sure to set your secret key. You should have it ready from step 1
```
Comment (Basharath) [inline README.md]: At this we don't need to mention about secret key as it is already mentioned before.

File: src/Code.js [hunk 3]
```diff
@@ -89,6 +99,44 @@ function action(req) {
...
+
+        if (!siteKey) {
+          response = {
+            status: 'error',
+            message: 'reCAPTCHA verification under key \'g-recaptcha-response\' is required.',
+          };
+          return ContentService.createTextOutput(JSON.stringify(response)).setMimeType(ContentService.MimeType.JSON);
+        }
+
+        var captchaResponse = UrlFetchApp.fetch('https://www.google.com/recaptcha/api/siteverify', {
```
Comment (Basharath) [inline src/Code.js]: Maybe you can use `const` as it is used every where.

File: README.md (hunk 1 above)
Comment (Meldiron) [reply README.md]: Updated, thanks!

File: README.md (hunk 2 above)
Comment (Meldiron) [reply README.md]: Applied. Also removed `> Make sure to set your secret key. You should have it ready from step 1` below step 4

File: src/Code.js (hunk 3 above)
Comment (Meldiron) [reply src/Code.js]: Hoopsie, copy&paste from docs. Nice catch.

File: README.md (hunk 2 above)
Comment (Basharath) [reply README.md]: I think you have removed this `> Make sure to set your secret key. You should have it ready from step 1` under step 2 as well. 
Keeping it below step 2 will give details about it. You can add it or after merging I'll include it.


So everything seems to be fine, when do you want me to merge and close this PR as you said you want to add more.

**Comment by Meldiron** [comment]
Hey @Basharath 👋
What do you think about PR - all good? Can I start implementing more adapters to support more captcha providers?

**Comment by Basharath** [comment]
Hey @Meldiron 
I was waiting to hear from you about that. 

It is all good so far, I haven't merged yet. 
Please add all the providers you want and then tell me so we can finalize the merge. Thanks!

**Comment by Meldiron** [comment]
Awesome, I'll do so in the upcoming days. Ill also take my time with research to see what Captcha providers are most commonly used.

**Comment by Basharath** [comment]
Yeah, that's fine, you can take your time. Once done just update here.

**Comment by scottydelta** [comment]
any ETA on merging of this PR? 

I am looking to use FormEasy with reCAPTCHA. Without CAPTCHA, it's just too much spam.

**Comment by Meldiron** [comment]
I won't be able to research other providers any time soon. With that said, reCAPTCHA is properly implemented and tested. This PR can be merged, in my opinion. In the future, I will open a new PR to add new providers.

Let me ping @Basharath about this.

**Comment by Basharath** [comment]
Hey, @Meldiron I thought to ask you about this a few days ago, meanwhile, this happened. 

I'll try merging this by this weekend and publish the new update.

**Pull request merged**